bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/registry/AbstractRegistry.java

```diff
@@ -268,6 +279,15 @@ protected void addProvider(Provider<E> provider) {
         }
     }
 
+    /**
+     * Returns the key for a given element
+     *
+     * @param element
+     *            element
+     * @return key (must not be null)
+     */
+    protected abstract K getKey(E element);
```

Comment:
Ah, sorry, I was completely mislead by the name %&$**Provider**, so just what @sjka already mentioned as the naming conflict.
Ok, then I agree that we should have a common interface on our entities. It seems that most people simply use an `Identifiable` interface, wouldn't that be fine here as well? Note that "UniquelyIdentifiable" might be rather misleading, because the ids are only unique within their own type, but not across all Identifiables!